### ISSUE TITLE
failing requests to examples.obspy.org

### ISSUE THREAD

**Issue by megies** [obspy/obspy#1599]
Especially in CI we have plenty of non-successful requests to examples.obspy.org somehow, which leads to lots of errors in CI tests..

E.g.: http://tests.obspy.org/60454/#1

```
```Traceback (most recent call last):
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/obspy/core/tests/test_stream.py", line 1720, in test_read
tr = read('https://examples.obspy.org/test.sac', dtype=np.int32)[0]
File "<decorator-gen-31>", line 2, in read
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/obspy/core/util/decorator.py", line 294, in _map_example_filename
return func(*args, **kwargs)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/obspy/core/stream.py", line 225, in read
download_to_file(url=pathname_or_url, filename_or_buffer=fh)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/obspy/core/util/base.py", line 602, in download_to_file
r = requests.get(url, stream=True)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/api.py", line 70, in get
return request('get', url, params=params, **kwargs)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/api.py", line 56, in request
return session.request(method=method, url=url, **kwargs)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/sessions.py", line 369, in __exit__
self.close()
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/sessions.py", line 690, in close
v.close()
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/adapters.py", line 306, in close
self.poolmanager.clear()
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/packages/urllib3/poolmanager.py", line 156, in clear
self.pools.clear()
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/packages/urllib3/_collections.py", line 94, in clear
self.dispose_func(value)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/packages/urllib3/poolmanager.py", line 117, in <lambda>
dispose_func=lambda p: p.close())
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 423, in close
conn = old_pool.get(block=False)
File "/home/travis/miniconda/envs/test-environment/lib/python2.7/Queue.py", line 165, in get
raise Empty
Empty

**Comment by megies** [comment]
I'm able to reproduce this problem on Python 2.7 with requests version 2.12.1. For me installing that requests version in the root environment even makes conda unusable as any conda install command fails due to exceptions when downloading package metadata. The only solution was to manually download an older requests conda package from Anaconda cloud and install it with `conda install ./requests....tar.bz2`.

To reproduce:

```bash
$ conda create -n requests_bug python=2.7 requests=2.12.1 obspy
$ source activate requests_bug
$ obspy-runtests core
```

Then to get rid of the error:
```bash
$ conda install 'requests<2.12'
$ obspy-runtests core
```

Maybe we should set our requests dependency to `<2.12` in setup.py and for conda package for now?

**Comment by krischer** [comment]
2.12.2 has been released yesterday. Does it solve the issue?

https://github.com/kennethreitz/requests/commit/ca15d4808734c86801c8f3d80c9152c35a163dc3

**Comment by megies** [comment]
Nope, still has the same problem.

**Comment by barsch** [comment]
is it an issue with the https cert? does it work for http links?

**Comment by megies** [comment]
No, also fails if I replace all `https://examples.obspy.org` with `http://` in `core/tests/test_stream.py`.

**Comment by barsch** [comment]
ok, just a wild guess - welcome to the dependency hell - another reason to keep the number of external dependencies low ...

anyway, I'm relieved its not some disk issue with the server

**Comment by krischer** [comment]
Really strange - I can also reproduce it on OSX. The really weird part is that it works if one just executes the offending statement. It only fails in the tests. This works fine for me:

```bash
python -c "import obspy; import numpy as np; print obspy.read('https://examples.obspy.org/test.sac', dtype=np.int32)[0]"
```

So I guess this is our fault somewhere?

Regarding keeping dependencies low: In general I totally agree but requests is kind of de-facto standard library and if any package is safe to use it should be it.

**Comment by megies** [comment]
See kennethreitz/requests#3742, it's an unlucky interaction of `six` and `future`. Don't ask me what's going on exactly, I'm just happy this can be fixed rather easily.

**Comment by krischer** [comment]
Man...we really have to get rid of `future`...

**Comment by krischer** [comment]
Thanks for digging into this!

**Comment by QuLogic** [comment]
> Man...we really have to get rid of future...

You meant to say Python 2, there... :wink:

**Comment by krischer** [comment]
> You meant to say Python 2, there... 😉

;-)

**Comment by megies** [comment]
Fixed by 48162a8c17db0337c835b5d97d0bc0c9577fc329 in #1612.
